### ISSUE TITLE
Downloading vs building from source code

### ISSUE THREAD

**Issue by markxtji** [sscarduzio/elasticsearch-readonlyrest-plugin#368]
Hello @sscarduzio ,

I am going to install  elasticsearch-readonlyrest-plugin  in docker.  But the way of downloading it becomes an obstacle because I have to provide you my email and the url in  the email is not permanent. 

Because my project is also open source, I am not sure if you comfortable for me to download the plugin from my email and put it in my project ?

If you are not comfortable with the above, then I will have to build it from your source code. But I cannot find the instruction in the readme file regarding how to build it.  Can you please advise ?

thank you.

**Comment by markxtji** [comment]
I found your instruction on how to build it.
https://github.com/beshu-tech/readonlyrest-docs/blob/master/elasticsearch.md
But it is quite hard to build it inside a docker container because I will need to install docker inside docker container, and a lot of other tools, and when the installation is done remove all these tools.

it is better if you are oK for me to put the zip file in my opensource project, or provide a permanent link to download it.

thank you.

**Comment by HagK** [comment]
Am 2018-06-07 05:14, schrieb markxtji:

> If you are not comfortable with the above, then I will have to build it from your source code. But I cannot find the instruction in the readme file regarding how to build it. Can you please advise ?

here is my good old guidline, tested with 1.16.20_pre  some days ago 

build readonlyrest plugin 

* download a release from github 

* extract the file in $WORKDIR e.g. /tmp or whatever 

* ensure you have docker installed and running  - test: docker ps 

* edit settings.gradle comment out include 'integration-tests'  and
maybe unused versions 

* edit esVERSION/gradle.properties and choose the right minor version 

* bin/build.sh 

* capture your plugin.zip  from esVERSION/build/distributions/ 

##  integration-tests  failed often, some months ago, so i have skipped
it 

regards 

Hagen

**Comment by sscarduzio** [comment]
@HagK  is right, you don't need to run tests, the version you want is well tagged in git and all the tests are run by Travis-CI already. No docker-in-docker necessary.

**Comment by markxtji** [comment]
hi @sscarduzio and @HagK ,

thank you for the instruction.

I commented inetegration test, but I still got below error. It is trying to find docker environment.

tech.beshu.ror.settings.LdapSettingsTests > classMethod STARTED

tech.beshu.ror.settings.LdapSettingsTests > classMethod FAILED
    java.lang.ExceptionInInitializerError

        Caused by:
        java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration
            at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$3(DockerClientProviderStrategy.java:155)
            at java.util.Optional.orElseThrow(Optional.java:290)
            at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:147)
            at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:99)
            at org.testcontainers.containers.GenericContainer.<init>(GenericContainer.java:142)
            at tech.beshu.ror.utils.containers.LdapContainer.<init>(LdapContainer.java:58)
            at tech.beshu.ror.utils.containers.LdapContainer.create(LdapContainer.java:66)
            at tech.beshu.ror.settings.LdapSettingsTests.<clinit>(LdapSettingsTests.java:31)

tech.beshu.ror.settings.LdapAuthorizationRuleSettingsTests > testRuleSettingsCreationFailsDueToNotSetLdapAuthName STARTED

**Comment by markxtji** [comment]
Hi @sscarduzio ,

I tried  ./gradlew --stacktrace es61x:ror '-PesVersion=6.1.1',

then I got an error like the following :

FAILURE: Build failed with an exception.

* Where:
Build file '/tmp/readonlyrest/core/build.gradle' line: 75

* What went wrong:
A problem occurred evaluating project ':core'.
> Project with path ':tests-utils' could not be found in project ':core'.

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.GradleScriptException: A problem occurred evaluating project ':core'.
        at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:92)


I commented line 75 in 
/tmp/readonlyrest/core/build.gradle

     73     compile 'org.apache.httpcomponents:httpasyncclient:4.1.3'
     74 
     75     //testCompile project(':tests-utils')
     76     testCompile 'org.gradle:gradle-tooling-api:2.13'

and run it again.
 ./gradlew --stacktrace es61x:ror '-PesVersion=6.1.1'

Build was successful.

Thank you very much 👍 

thanks @HagK  too :-)
